### PR TITLE
Disable editing in support when subsequent application exists

### PR DESCRIPTION
**Pull request by benswannack**
## Context
We currently allow editing of details on any application form in support. When a candidate has an apply 1 form followed by an apply 2 form, we generally want to only edit the latest entry as their old form is no longer relevant.

## Changes proposed in this pull request
Hide edit links in support on an application form when the application is not the latest one in the chain.

Add a banner to the application page in support to explain why the application is no longer editable
Screenshot 2021-09-30 at 09 24 28

## Guidance to review
Per commit.
Does the logic make sense?
Are there any cases I've missed here?

## Link to Trello card
https://trello.com/c/UNpmwpgT/4287-prevent-changes-being-made-to-an-applicants-application-in-apply1-when-they-have-an-application-in-apply2

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
Looking good, I think we missed work experience, reverting rejections and undoing refusals of referees, could we include them as well?

**Comment by benswannack** [comment]
> Looking good, I think we missed work experience, reverting rejections and undoing refusals of referees, could we include them as well?

Good spot! As discussed I'll raise a card for the work experience one as it's currently broken in a different way (the change links go to the candidate interface). The others I'll fix here.

**Review by cwrw — APPROVED** [review]
like the banner

File: app/views/support_interface/application_forms/_application_navigation.html.erb [hunk 1]
```diff
@@ -20,6 +20,15 @@
   </div>
 </div>
 
+<% unless @application_form.editable? %>
+  <%= govuk_notification_banner(title_text: t('notification_banner.info')) do |notification_banner| %>
+    <% notification_banner.heading(text: 'Candidate has a subsequent application') %>
```
Comment (benswannack) [inline app/views/support_interface/application_forms/_application_navigation.html.erb:25]: Discussed and agreed that "more recent" is a bit clearer than "subsequent"

**Pull request merged**